src/address.c

```diff
@@ -393,3 +394,174 @@ int getDerivedHDAddress(const char* masterkey, uint32_t account, bool ischange,
         int ret = getDerivedHDAddressByPath(masterkey, derived_path, outaddress, outprivkey);
         return ret;
 }
+
+/**
+ * @brief This function generates a new dogecoin address from a mnemonic by the slip44 key path.
+ *
+ * @param account The BIP44 account to generate the derived address.
+ * @param index The BIP44 index to generate the derived address.
+ * @param change_level The BIP44 change level flag to generate derived address.
+ * @param mnemonic The mnemonic code words.
+ * @param passphrase The passphrase (optional).
+ * @param p2pkh_pubkey_master The generated master public key.
+ * @param is_testnet The flag denoting which network, 0 for mainnet and 1 for testnet.
+ *
+ * return: 0 (success), -1 (fail)
+ */
+int getDerivedHDAddressFromMnemonic(const uint32_t account, const uint32_t index, const CHANGE_LEVEL change_level, const MNEMONIC mnemonic, const PASSPHRASE pass, char* p2pkh_pubkey, const dogecoin_bool is_testnet) {
+
+    /* Validate input */
+    if (!mnemonic) {
+        fprintf(stderr, "ERROR: Invalid mnemonic\n");
+        return -1;
+    }
+
+    /* Initialize variables */
+    dogecoin_hdnode node;
+    dogecoin_hdnode bip44_key;
+    char keypath[BIP44_KEY_PATH_MAX_LENGTH + 1] = "";
+
+    /* Allocate memory for seed and initialize */
+    uint8_t* seed = malloc(64);
```

Comment:
Can this just be on the stack as `uint8_t seed[MAX_SEED_SIZE]` or just `SEED seed` to avoid all the worry about allocation/freeing?